--- src/LinuxClient/application/iFolderApplication.cs	2006-08-02 01:24:36.000000000 -0600
+++ src/LinuxClient/application/iFolderApplication.cs.new	2006-09-11 11:34:43.000000000 -0600
@@ -65,7 +65,7 @@
 		private Gdk.Pixbuf			DownloadingPixbuf;
 		private Gdk.Pixbuf			UploadingPixbuf;
 		private Gtk.EventBox		eBox;
-		private TrayIcon			tIcon;
+		private Egg.TrayIcon			tIcon;
 		private iFolderWebService	ifws;
 		private SimiasWebService	simws;
 		private iFolderData			ifdata;
@@ -141,7 +141,7 @@
 			
 			Util.SetQuitiFolderDelegate(new QuitiFolderDelegate(QuitiFolder));
 
-			tIcon = new TrayIcon("iFolder");
+			tIcon = new Egg.TrayIcon("iFolder");
 
 			currentIconAnimationDirection = 0;
 
