--- Makefile.orig	2009-04-26 07:26:32.128227827 -0600
+++ Makefile	2009-04-26 07:27:07.220196493 -0600
@@ -19,7 +19,7 @@
 
 	# run icon-naming-utils
 	for d in $$DESTDIR/usr/share/icons/Human/*; do \
-	    (cd $$d; for c in *; do /usr/lib/icon-naming-utils/icon-name-mapping -c $$c; done); \
+	    (cd $$d; for c in *; do /usr/libexec/icon-name-mapping -c $$c; done); \
 	done
 
 	# install mo files
