--- configure.in	2007-08-18 00:53:11.000000000 -0600
+++ configure.in.new	2007-08-18 01:19:24.000000000 -0600
@@ -279,7 +279,7 @@
         AC_CHECK_PROG(CXX, g++, g++)
 		StoreProviderAssembly="SimiasLib.dll"
 		StoreProviderType="Simias.Storage.Provider.Flaim.FlaimProvider"
-		GSOAP_PATH="$TOOLDIR/gsoap/linux-2.7"
+		GSOAP_PATH="$TOOLDIR/gsoap/gsoap-linux-2.7"
 		mappingpath="${exec_prefix}/lib/simias/web/bin/Simias.exe"
     ;;
     #
