--- src/core/libsimias/Makefile.am	2007-08-18 01:51:50.000000000 -0600
+++ src/core/libsimias/Makefile.am.new	2007-08-18 01:51:35.000000000 -0600
@@ -27,6 +27,9 @@
 	$(GSOAP_PATH)/soapcpp2 -c -penv $(srcdir)/env.h
 
 $(srcdir)/stdsoap2.c:
+	cd $(GSOAP_PATH)/src && make && cp soapcpp2 $(GSOAP_PATH)
+	cd $(GSOAP_PATH)/wsdl && make
+	cp $(GSOAP_PATH)/bin/* $(GSOAP_PATH)
 	cp $(GSOAP_PATH)/stdsoap2.h $(srcdir)
 	cp $(GSOAP_PATH)/stdsoap2.c $(srcdir)
 
