--- src/server/setup/iFolderWebSetup.cs	2007-10-29 18:30:06.000000000 -0600
+++ src/server/setup/iFolderWebSetup.cs.new	2008-01-24 12:14:10.000000000 -0700
@@ -155,6 +155,9 @@
 			{
 				// ignore
 			}
+
+			apacheUser = "apache";
+			apacheGroup = "apache";
 		}
 
 		/// <summary>
@@ -325,7 +328,7 @@
 		/// </summary>
 		void SetupModMono()
 		{
-			string path = "/etc/apache2/conf.d/ifolder_web.conf";
+			string path = "/etc/simias/apache/gentoo/ifolder_web.conf";
 
 			Console.Write("Configuring {0}...", path);
 
@@ -355,8 +358,6 @@
                     sslPrefix = "#";
                 }
 
-				writer.WriteLine("Include /etc/apache2/mod_mono.conf");
-				writer.WriteLine();
 				writer.WriteLine("Alias /{0} \"{1}\"", alias, webPath);
 				writer.WriteLine("AddMonoApplications {0} \"/{0}:{1}\"", alias, webPath);
                                 if (logoutUrl.Value != String.Empty || logoutUrl.Value.Trim () != "" )
