--- src/server/setup/iFolderAdminSetup.cs	2007-10-29 18:30:06.000000000 -0600
+++ src/server/setup/iFolderAdminSetup.cs.new	2008-01-24 12:10:56.000000000 -0700
@@ -154,6 +154,10 @@
 			{
 				// ignore
 			}
+
+			apacheUser = "apache";
+			apacheGroup = "apache";
+
 		}
 
 		/// <summary>
@@ -324,7 +328,7 @@
 		/// </summary>
 		void SetupModMono()
 		{
-			string path = "/etc/apache2/conf.d/ifolder_admin.conf";
+			string path = "/etc/simias/apache/gentoo/ifolder_admin.conf";
 
 			Console.Write("Configuring {0}...", path);
 
@@ -354,8 +358,6 @@
                     sslPrefix = "#";
                 }
 
-				writer.WriteLine("Include /etc/apache2/mod_mono.conf");
-				writer.WriteLine();
 				writer.WriteLine("Alias /{0} \"{1}\"", alias, webPath);
 				writer.WriteLine("AddMonoApplications {0} \"/{0}:{1}\"", alias, webPath);
 
