--- src/server/setup/SimiasServerSetup.cs	2007-01-02 01:45:42.000000000 -0700
+++ src/server/setup/SimiasServerSetup.cs.new	2007-01-03 14:48:05.000000000 -0700
@@ -81,12 +81,12 @@
 		/// <summary>
 		/// Apache User
 		/// </summary>
-		string apacheUser = "wwwrun";
+		string apacheUser = "apache";
 
 		/// <summary>
 		/// Apache Group
 		/// </summary>
-		string apacheGroup = "www";
+		string apacheGroup = "apache";
 
 		/// <summary>
 		/// The uri to the ldap server.
