diff -Naur simias/src/server/setup/Makefile.am simias-changed/src/server/setup/Makefile.am
--- simias/src/server/setup/Makefile.am	2007-10-01 22:53:18.000000000 +0200
+++ simias-changed/src/server/setup/Makefile.am	2007-10-02 08:52:47.000000000 +0200
@@ -120,9 +120,9 @@
 	$(INSTALL_PROGRAM) ifolder-web-setup $(DESTDIR)$(bindir)
 	$(INSTALL_PROGRAM) ifolder-admin-setup $(DESTDIR)$(bindir)
 	$(INSTALL_PROGRAM) utility.exe $(DEBUG_FILES) $(DESTDIR)$(bindir)
-	$(mkinstalldirs) /etc/ld.so.conf.d
-	$(INSTALL_PROGRAM) novell-ifolder3.conf /etc/ld.so.conf.d
-	grep -q "include /etc/ld.so.conf.d/\*.conf" /etc/ld.so.conf || echo "include /etc/ld.so.conf.d/*.conf" >> /etc/ld.so.conf
+#	$(mkinstalldirs) /etc/ld.so.conf.d
+#	$(INSTALL_PROGRAM) novell-ifolder3.conf /etc/ld.so.conf.d
+#	grep -q "include /etc/ld.so.conf.d/\*.conf" /etc/ld.so.conf || echo "include /etc/ld.so.conf.d/*.conf" >> /etc/ld.so.conf
 
 	
 endif
