--- src/server/setup/SimiasServerSetup.cs	2007-10-29 18:30:06.000000000 -0600
+++ src/server/setup/SimiasServerSetup.cs.new	2008-01-24 12:22:21.000000000 -0700
@@ -580,6 +580,9 @@
 			        apacheUser.Prompt = apacheGroup.Prompt = true;
 			}
 
+			apacheUser.Value = "apache";
+			apacheGroup.Value = "apache";
+
 		}
 
 		bool Configure()
@@ -1143,7 +1146,7 @@
 		/// </summary>
 		void SetupModMono()
 		{
-			string path = Path.GetFullPath( "/etc/apache2/conf.d/simias.conf" );
+			string path = Path.GetFullPath( "/etc/simias/apache/gentoo/simias.conf" );
 			Console.Write("Configuring {0}...", path);
 
 			if ( apache.Value == true )
@@ -1167,8 +1170,6 @@
 				
 					string alias = "simias10";
 
-					writer.WriteLine( "Include /etc/apache2/mod_mono.conf" );
-					writer.WriteLine();
 					writer.WriteLine("Alias /{0} \"{1}\"", alias, SimiasSetup.webdir);
 					writer.WriteLine("AddMonoApplications {0} \"/{0}:{1}\"", alias, SimiasSetup.webdir);
 					writer.WriteLine("MonoSetEnv {0} \"SimiasRunAsServer=true;SimiasDataDir={1}\"", alias, this.storePath);
