--- src/webaccess/Browse.aspx.old	2006-05-09 12:27:13.000000000 -0600
+++ src/webaccess/Browse.aspx	2006-05-09 12:30:06.000000000 -0600
@@ -2,8 +2,6 @@
 <%@ Register TagPrefix="iFolder" TagName="HeaderControl" Src="Header.ascx" %>
 <%@ Register TagPrefix="iFolder" TagName="MessageControl" Src="Message.ascx" %>
 <%@ Register TagPrefix="iFolder" TagName="iFolderContextControl" Src="iFolderContext.ascx" %>
-<%@ Register TagPrefix="iFolder" TagName="TabControl" Src="TabControl.ascx" %>
-<%@ Register TagPrefix="iFolder" TagName="QuotaControl" Src="Quota.ascx" %>
 <%@ Register TagPrefix="iFolder" TagName="PaggingControl" Src="Pagging.ascx" %>
 <!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
 <html>
@@ -75,9 +73,6 @@
 	
 		<div id="nav">
 	
-			<iFolder:TabControl id="Tabs" runat="server" />
-	
-			<iFolder:QuotaControl runat="server" />
 
 		</div>
 	
@@ -176,4 +171,4 @@
 
 </body>
 
-</html>
\ No newline at end of file
+</html>
