diff -r -u zeitgeist-sharp-0.1.0.3//Zeitgeist/Datamodel/DataSource.cs zeitgeist-sharp-0.1.0.3-patched//Zeitgeist/Datamodel/DataSource.cs
--- zeitgeist-sharp-0.1.0.3//Zeitgeist/Datamodel/DataSource.cs	2010-11-01 08:02:21.000000000 -0600
+++ zeitgeist-sharp-0.1.0.3-patched//Zeitgeist/Datamodel/DataSource.cs	2010-12-14 01:36:42.332000043 -0700
@@ -229,13 +229,13 @@
 		/// </param>
 		public RawDataSource(string uid, string name, string desc, RawEvent[] events, bool running, UInt64 Lastseen, bool enabled)
 		{
-			UniqueId = uid;
-			Name = name;
-			Description = desc;
-			RawEvents = events;
-			Running = running;
-			LastSeen = Lastseen;
-			Enabled = enabled;
+			this.UniqueId = uid;
+			this.Name = name;
+			this.Description = desc;
+			this.RawEvents = events;
+			this.Running = running;
+			this.LastSeen = Lastseen;
+			this.Enabled = enabled;
 		}
 		
 		/// <summary>
Only in zeitgeist-sharp-0.1.0.3-patched//Zeitgeist: Makefile
diff -r -u zeitgeist-sharp-0.1.0.3//Zeitgeist/Zeitgeist.csproj zeitgeist-sharp-0.1.0.3-patched//Zeitgeist/Zeitgeist.csproj
--- zeitgeist-sharp-0.1.0.3//Zeitgeist/Zeitgeist.csproj	2010-11-01 05:47:52.000000000 -0600
+++ zeitgeist-sharp-0.1.0.3-patched//Zeitgeist/Zeitgeist.csproj	2010-12-14 01:29:24.114000039 -0700
@@ -32,7 +32,8 @@
     <ConsolePause>false</ConsolePause>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="System" />
+    <Reference Include="System, Version=2.0.0.0" />
+    <Reference Include="System.Core, Version=3.5.0.0" />
     <Reference Include="NDesk.DBus, Version=1.0.0.0, Culture=neutral, PublicKeyToken=f6716e4f9b2ed099">
       <Package>ndesk-dbus-1.0</Package>
     </Reference>
@@ -70,4 +71,4 @@
     <Folder Include="Client\" />
   </ItemGroup>
   <Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />
-</Project>
\ No newline at end of file
+</Project>
Only in zeitgeist-sharp-0.1.0.3-patched//Zeitgeist: zeitgeist-sharp.pc
