--- gnome-panel-2.12.2/gnome-panel/panel-recent.c
+++ gnome-panel-2.12.2/gnome-panel/panel-recent.c
@@ -45,10 +45,22 @@
 	  GError **error)
 {
 	char **env;
+	char *local_path;
 	GnomeVFSResult result;
 	GnomeVFSMimeApplication *app;
 	GList *uris = NULL;
 
+	local_path = gnome_vfs_get_local_path_from_uri (uri);
+	
+	if (local_path && !g_file_test (local_path, G_FILE_TEST_EXISTS)) {
+		g_set_error (error, 0, 0,
+			     _("%s does not exist."), local_path);
+		g_free (local_path);
+		return FALSE;
+	}
+	g_free (local_path);
+
+
 	app = gnome_vfs_mime_get_default_application_for_uri (uri, mime_type);
 	if (app == NULL) {
 		g_set_error (error, 0, 0, _("Couldn't find a suitable application"));
