--- /usr/share/php/MDB2/Driver/mysqli.php	2010-02-10 16:42:04.059261209 -0700
+++ ./mysqli.php	2010-03-04 19:26:01.512040111 -0700
@@ -1738,7 +1738,12 @@
             if (!is_object($this->statement)) {
                 $query.= ' USING @'.implode(', @', array_values($this->positions));
             } else {
-                $result = @call_user_func_array('mysqli_stmt_bind_param', $parameters);
+
+                foreach ($parameters as $key => $value) {
+                    $reference_parameters_hack[$key] = &$parameters[$key];
+                }
+                $result = @call_user_func_array('mysqli_stmt_bind_param', $reference_parameters_hack);
+                //$result = @call_user_func_array('mysqli_stmt_bind_param', $parameters);
                 if ($result === false) {
                     $err =& $this->db->raiseError(null, null, null,
                         'Unable to bind parameters', __FUNCTION__);
@@ -1846,4 +1851,4 @@
         return $result;
    }
 }
-?>
\ No newline at end of file
+?>
