--- ubuntulooks-0.9.12.orig/themes/Human/gtk-2.0/gtkrc
+++ ubuntulooks-0.9.12/themes/Human/gtk-2.0/gtkrc
@@ -7,6 +7,7 @@
 #
 # Feel free to modify and share!
 
+gtk-icon-sizes = "panel-menu=24,24"
 
 style "clearlooks-default"
 {
@@ -219,6 +220,7 @@
 # tooltips stuff
 widget_class "*.tooltips.*.GtkToggleButton" style "clearlooks-tasklist"
 widget "gtk-tooltips" style "clearlooks-tooltips"
+widget "gtk-tooltip" style "clearlooks-tooltips"
 
 # treeview stuff
 widget_class "*.GtkTreeView.GtkButton" style "clearlooks-tree"
--- ubuntulooks-0.9.12.orig/debian/patches/01_fix_listview_arrows_drawing.patch
+++ ubuntulooks-0.9.12/debian/patches/01_fix_listview_arrows_drawing.patch
@@ -0,0 +1,13 @@
+diff -Nur ubuntulooks-0.9.12/engine/src/ubuntulooks_style.c ubuntulooks-0.9.12.new/engine/src/ubuntulooks_style.c
+--- ubuntulooks-0.9.12/engine/src/ubuntulooks_style.c	2006-08-23 12:38:26.000000000 +0200
++++ ubuntulooks-0.9.12.new/engine/src/ubuntulooks_style.c	2007-02-12 17:28:36.000000000 +0100
+@@ -1084,6 +1084,9 @@
+ 		
+ 	sanitize_size (window, &width, &height);
+ 
++	if (arrow_type == (GtkArrowType)4)
++	  return;
++
+ 	if (DETAIL ("arrow"))
+ 	{
+ 		WidgetParameters params;
--- ubuntulooks-0.9.12.orig/debian/patches/01_fix_tick_box_drawing.patch
+++ ubuntulooks-0.9.12/debian/patches/01_fix_tick_box_drawing.patch
@@ -0,0 +1,87 @@
+diff -Nur ubuntulooks-0.9.12/engine/src/ubuntulooks_draw.c ubuntulooks-0.9.12.new/engine/src/ubuntulooks_draw.c
+--- ubuntulooks-0.9.12/engine/src/ubuntulooks_draw.c	2006-09-01 18:57:20.000000000 +0200
++++ ubuntulooks-0.9.12.new/engine/src/ubuntulooks_draw.c	2006-10-09 16:26:29.000000000 +0200
+@@ -1837,14 +1837,6 @@
+ 	cairo_pattern_t *pattern;
+ 	gboolean draw_bullet = (checkbox->shadow_type == GTK_SHADOW_IN);
+ 
+-	/* sanitize size to be rectangular */
+-	if (width > height) {
+-		x += (width - height) / 2;
+-		width = height;
+-	} else if (height > width) {
+-		y += (height - width) / 2;
+-		height = width;
+-	}
+ 	
+ 	cairo_set_line_width (cr, 1.0);
+ 	
+@@ -1947,8 +1939,6 @@
+ 		
+ 	}
+ 	
+-	cairo_translate (cr, x, y);
+-	cairo_scale (cr, width / 13.0, height / 13.0);
+ 	/* To offset this checkmark, one should probably use a translation */
+ 	if (draw_bullet)
+ 	{
+@@ -1961,7 +1951,7 @@
+ 			cairo_translate (cr, -2, -2);
+ 		}
+ 	
+-		cairo_move_to (cr, 4, 8);
++		cairo_move_to (cr, x + 4, y + 8);
+ 		cairo_rel_line_to (cr,   5,   4);
+ 		cairo_rel_curve_to (cr,  1.4,  -5,   -1,  -1,   5.7,  -12.5);
+ 		cairo_rel_curve_to (cr, -4,   4,  -4,   4,  -6.7,    9.3);
+@@ -1975,7 +1965,10 @@
+ 	}
+ 	else if (checkbox->shadow_type == GTK_SHADOW_ETCHED_IN)
+ 	{
+-		cairo_rectangle (cr, 4.0, 6.0, 6, 2);
++		int c_x = x + (width/2) - 2;
++		int c_y = y + (height/2);
++
++		cairo_rectangle (cr, c_x, c_y, 6, 2);
+ 		cairo_set_source_rgb (cr, colors->text[widget->state_type].r,
+ 		                          colors->text[widget->state_type].g,
+ 		                          colors->text[widget->state_type].b);
+@@ -1993,15 +1986,7 @@
+ 	CairoColor border, top, bottom;
+ 	cairo_pattern_t *pattern;
+ 	gboolean draw_bullet = (checkbox->shadow_type == CL_SHADOW_IN || checkbox->shadow_type == CL_SHADOW_ETCHED_IN);
+-	/* sanitize size to be rectangular */
+-	if (width > height) {
+-		x += (width - height) / 2;
+-		width = height;
+-	} else if (height > width) {
+-		y += (height - width) / 2;
+-		height = width;
+-	}
+-
++	
+ 	cairo_set_line_width (cr, 1.0);
+ 	
+ 	if (checkbox->in_menu)
+@@ -2085,11 +2070,9 @@
+ 	cairo_stroke (cr);
+ 	
+ 	// draw the bullet
+-	cairo_translate (cr, x, y);
+-	cairo_scale (cr, width / 13.0, height / 13.0);
+ 	if (draw_bullet)
+ 	{
+-		cairo_arc (cr, 6.5, 6.5, 2.5, 0, 2 * M_PI);
++		cairo_arc (cr, x+width/2., y+height/2., 2.5, 0, 2 * M_PI);
+ 		if (widget->disabled)
+ 		{
+ 			cairo_set_source_rgb (cr, colors->text[widget->state_type].r,
+@@ -2098,7 +2081,7 @@
+ 		}
+ 		else
+ 		{
+-			pattern = cairo_pattern_create_radial (5.25, 5.25, 0.1, 5.25, 5.25, 4);
++			pattern = cairo_pattern_create_radial (x+width/2. - 1.25, y+height/2. - 1.25, 0.1, x+width/2. - 1.25, y+height/2. - 1.25, 4);
+ 			cairo_pattern_add_color_stop_rgb (pattern, 0.0, 0.4, 0.4, 0.4);
+ 			cairo_pattern_add_color_stop_rgb (pattern, 1.0, colors->text[widget->state_type].r,
+ 			                                                colors->text[widget->state_type].g,
--- ubuntulooks-0.9.12.orig/debian/copyright
+++ ubuntulooks-0.9.12/debian/copyright
@@ -0,0 +1,28 @@
+This package was debianized by Sebastien Bacher <seb128@canonical.com> on
+Tue,  7 Mar 2006 16:10:39 +0000.
+
+It was downloaded from 
+
+Copyright Holder:
+  Richard Stellingwerff - remenic@gmail.com
+  Daniel Borgmann       - daniel.borgmann@gmail.com
+
+License:
+
+   This package is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This package is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this package; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+
+On Debian systems, the complete text of the GNU General
+Public License can be found in `/usr/share/common-licenses/GPL'.
+
--- ubuntulooks-0.9.12.orig/debian/changelog
+++ ubuntulooks-0.9.12/debian/changelog
@@ -0,0 +1,146 @@
+ubuntulooks (0.9.12-5) gutsy; urgency=low
+
+  * themes/Human/gtk-2.0/gtkrc:
+     - update the tooltips definition for the new API, change suggested
+       by Alex Jones (LP: #129699)
+
+ -- Sebastien Bacher <seb128@canonical.com>  Tue, 07 Aug 2007 10:40:43 +0200
+
+ubuntulooks (0.9.12-4) feisty; urgency=low
+
+  * themes/Human/gtk-2.0/gtkrc:
+    - use 'gtk-icon-sizes = "panel-menu=24,24"' (Ubuntu: #43398)
+
+ -- Sebastien Bacher <seb128@canonical.com>  Wed,  4 Apr 2007 14:20:32 +0200
+
+ubuntulooks (0.9.12-3) feisty; urgency=low
+
+  * debian/patches/01_fix_listview_arrows_drawing.patch:
+    - fix all listview columns having an arrow drawing (Ubuntu: #71339)
+
+ -- Sebastien Bacher <seb128@canonical.com>  Mon, 12 Feb 2007 17:28:49 +0100
+
+ubuntulooks (0.9.12-2) edgy; urgency=low
+
+  * debian/patches/01_fix_tick_box_drawing.patch:
+    - undo patch from bug #57749, it make tick box being not correctly centered
+      (Ubuntu: #58610)
+
+ -- Sebastien Bacher <seb128@canonical.com>  Mon,  9 Oct 2006 16:26:46 +0200
+
+ubuntulooks (0.9.12-1) edgy; urgency=low
+
+  * New upstream version (UVF exception granted by Colin Watson):
+    - Fixed multithreaded locking issue. Backport from Clearlooks #342199.
+      (Ubuntu: #57040)
+    - Applied patch from launchpad #57749 to fix erroneous drawing of checkbox
+      and radiobutton at non rectangular sizes.
+
+ -- Sebastien Bacher <seb128@canonical.com>  Fri,  1 Sep 2006 19:44:35 +0200
+
+ubuntulooks (0.9.11-2) edgy; urgency=low
+
+  * debian/control:
+    - build with GTK 2.10
+
+ -- Sebastien Bacher <seb128@canonical.com>  Wed,  5 Jul 2006 13:05:27 +0200
+
+ubuntulooks (0.9.11-1) dapper; urgency=low
+
+  * New upstream version:
+    - Render GtkScale properly at small sizes.
+
+ -- Sebastien Bacher <seb128@canonical.com>  Mon, 15 May 2006 23:16:27 +0200
+
+ubuntulooks (0.9.10-1) dapper; urgency=low
+
+  * New upstream version:
+    - Don't draw grip dots on scrollbar if there's no room (Ubuntu: #37415).
+    - Draw only 6 dots on panel grip.
+    - Render inconsistent checkboxes properly in cells (Ubuntu: #42558).
+
+ -- Sebastien Bacher <seb128@canonical.com>  Mon, 15 May 2006 22:36:40 +0200
+
+ubuntulooks (0.9.9-1) dapper; urgency=low
+
+  * New upstream version:
+    - Changed prelight on radio/checkbutton
+    - Fixed bug that affected gedit (Ubuntu: #35327)
+    - Reverted back to the old color for most widgets.
+
+ -- Sebastien Bacher <seb128@canonical.com>  Tue, 28 Mar 2006 15:35:35 +0200
+
+ubuntulooks (0.9.8-1) dapper; urgency=low
+
+  * New upstream version:
+    - Fix titlebar color with compiz (Ubuntu: #34497)
+    - Speed up drawing tabs
+    - Cleaned up some code
+    - Stronger contrast on selected menuitem in GDM
+    - Rounded corners in checkbox/radiobutton prelight
+    - Changed the selection colors
+
+ -- Sebastien Bacher <seb128@canonical.com>  Thu, 16 Mar 2006 19:11:59 +0100
+
+ubuntulooks (0.9.7-1) dapper; urgency=low
+
+  * New upstream version
+    - Disabled scale widgets look more disabled
+    - Darker scrollbar trough
+    - Renamed shade() to ul_shade() to avoid symbol conflicts (Ubuntu: #34325)
+  
+ -- Sebastien Bacher <seb128@canonical.com>  Tue, 14 Mar 2006 17:03:16 +0100
+
+ubuntulooks (0.9.6-1) dapper; urgency=low
+
+  * New upstream version:
+    - tooltip gradient
+    - Fixed crasher in evolution calendar
+
+ -- Sebastien Bacher <seb128@canonical.com>  Sun, 12 Mar 2006 22:09:59 +0100
+
+ubuntulooks (0.9.5-1) dapper; urgency=low
+
+  * New upstream version:
+    - Changed the background selection color
+    - Changed the metacity theme color
+    - Disabled scrollbars look disabled
+    - Fixed progressbar color in menus
+
+ -- Sebastien Bacher <seb128@canonical.com>  Fri, 10 Mar 2006 16:20:42 +0000
+
+ubuntulooks (0.9.4-1) dapper; urgency=low
+
+  * New upstream version
+
+ -- Sebastien Bacher <seb128@canonical.com>  Fri, 10 Mar 2006 14:39:55 +0000
+
+ubuntulooks (0.9.3-1) dapper; urgency=low
+
+  * New upstream version:
+    - fix color for progress bar with firefox and thunderbird (#31480)
+
+ -- Sebastien Bacher <seb128@canonical.com>  Fri, 10 Mar 2006 12:55:07 +0000
+
+ubuntulooks (0.9.2-1) dapper; urgency=low
+
+  * debian/control:
+    - Replaces ubuntu-artwork (<< 7) for the gtkrc move
+  * debian/rules:
+    - ship the gtkrc with that package rather than ubuntu-artwork so 
+      we don't have to upload both at every change
+
+ -- Sebastien Bacher <seb128@canonical.com>  Fri, 10 Mar 2006 10:05:43 +0000
+
+ubuntulooks (0.9.0-2) dapper; urgency=low
+
+  * debian/control: Build-depend on libgtk2.0-dev (>= 2.8.0).
+
+ -- Colin Watson <cjwatson@ubuntu.com>  Thu,  9 Mar 2006 20:01:33 +0000
+
+ubuntulooks (0.9.0-1) dapper; urgency=low
+
+  *  First package
+
+ -- Sebastien Bacher <seb128@canonical.com>  Tue,  7 Mar 2006 16:10:39 +0000
+
--- ubuntulooks-0.9.12.orig/debian/control
+++ ubuntulooks-0.9.12/debian/control
@@ -0,0 +1,15 @@
+Source: ubuntulooks
+Section: gnome
+Priority: optional
+Maintainer: Sebastien Bacher <seb128@canonical.com>
+Build-Depends: cdbs, debhelper (>= 4.1.0), libgtk2.0-dev (>= 2.10.0)
+Standards-Version: 3.6.2
+
+Package: gtk2-engines-ubuntulooks
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Replaces: ubuntu-artwork (<< 7)
+Description: 'ubuntulooks' theme for GTK+ 2.x
+ Clearlooks is a modern-looking engine for GTK+, based on Bluecurve. It
+ features elements of various popular themes, blended together to produce a
+ theme that is easy on the eyes and visually pleasing.
--- ubuntulooks-0.9.12.orig/debian/rules
+++ ubuntulooks-0.9.12/debian/rules
@@ -0,0 +1,6 @@
+#!/usr/bin/make -f
+
+include /usr/share/cdbs/1/class/autotools.mk
+include /usr/share/cdbs/1/rules/debhelper.mk
+include /usr/share/cdbs/1/class/gnome.mk
+include /usr/share/cdbs/1/rules/simple-patchsys.mk
--- ubuntulooks-0.9.12.orig/debian/docs
+++ ubuntulooks-0.9.12/debian/docs
@@ -0,0 +1,2 @@
+NEWS
+README
--- ubuntulooks-0.9.12.orig/debian/compat
+++ ubuntulooks-0.9.12/debian/compat
@@ -0,0 +1 @@
+4
--- ubuntulooks-0.9.12.orig/debian/dirs
+++ ubuntulooks-0.9.12/debian/dirs
@@ -0,0 +1,2 @@
+usr/bin
+usr/sbin
