--- src/SimiasClient/SimiasSetup.cs.in	2006-04-06 23:54:46.000000000 -0600
+++ src/SimiasClient/SimiasSetup.cs.in.new	2006-04-06 23:54:19.000000000 -0600
@@ -117,9 +117,9 @@
 				_sbindir = Path.GetFullPath(_sbindir.Replace(_prefix, value));
 				_sharedstatedir = Path.GetFullPath(_sharedstatedir.Replace(_prefix, value));
 				_sysconfdir = Path.GetFullPath(_sysconfdir.Replace(_prefix, value));
-				_webdir = Path.GetFullPath(_webdir.Replace(_prefix, value));
-				_webbindir = Path.GetFullPath(_webbindir.Replace(_prefix, value));
-				_modulesdir = Path.GetFullPath(_modulesdir.Replace(_prefix, value));
+			//	_webdir = Path.GetFullPath(_webdir.Replace(_prefix, value));
+			//	_webbindir = Path.GetFullPath(_webbindir.Replace(_prefix, value));
+			//	_modulesdir = Path.GetFullPath(_modulesdir.Replace(_prefix, value));
 				
 				// update prefix
 				_prefix = Path.GetFullPath(value);
