--- src/FlaimProvider/FlaimWrapper/CSPStore.cpp	2006-08-02 01:11:34.000000000 -0600
+++ src/FlaimProvider/FlaimWrapper/CSPStore.cpp.new	2006-09-21 00:54:49.000000000 -0600
@@ -1037,7 +1037,7 @@
 
 				if (caseSensitive)
 				{
-					rc = FlmCursorSetMode(cursor, FLM_COMP_WILD);
+					rc = FlmCursorSetMode(cursor, FLM_WILD_TAG);
 				}
 				rc = FlmCursorAddField(cursor, fieldId, 0);
 				if (RC_OK(rc))
