--- control-center/etc/control-center.schemas.in	2006-05-21 19:54:09.000000000 -0600
+++ control-center/etc/control-center.schemas.in.new	2006-06-24 10:09:47.000000000 -0600
@@ -7,7 +7,7 @@
 			<type>list</type>
 			<list_type>string</list_type>
 			<locale name="C">
-				<default><!-- Translators: The format of this string is the task name to be displayed (translate that part) followed by a ";" separator then the filename (DONT translate the file name) of a .desktop file to launch. Multiple entries are separated by a "," -->[Change Desktop Background;background.desktop,Add Printer;gnome-cups-manager.desktop,Configure Network;YaST2/lan.desktop,Change Password;gnome-passwd.desktop,Add User;YaST2/users.desktop,Open Administrator Settings;YaST.desktop]</default>
+				<default><!-- Translators: The format of this string is the task name to be displayed (translate that part) followed by a ";" separator then the filename (DONT translate the file name) of a .desktop file to launch. Multiple entries are separated by a "," -->[Change Desktop Background;background.desktop,Screensaver Settings;gnome-screensaver-preferences.desktop,Power Preferences;gnome-power-preferences.desktop,Manage Printers;gnome-cups-manager.desktop,Add User;users.desktop]</default>
 				<short>Task names and associated .desktop files</short>
 				<long>The task name to be displayed in the control-center (thus needing to be translated) followed by a ";" separator then the filename of an associated .desktop file to launch for that task.</long>
 			</locale>
