--- libslab/slab-gnome-util.c	2006-11-01 11:25:43.000000000 -0700
+++ libslab/slab-gnome-util.c.new	2006-12-18 16:44:54.000000000 -0700
@@ -170,34 +170,8 @@
 	gint retval;
 	GError *error;
 
-	argv = g_new (gchar *, 6);
-	argv[0] = "rpm";
-	argv[1] = "-qf";
-	argv[2] = "--qf";
-	argv[3] = "%{NAME}";
-	argv[4] = g_filename_from_uri (gnome_desktop_item_get_location (desktop_item), NULL, NULL);
-	argv[5] = NULL;
+	return g_filename_from_uri (gnome_desktop_item_get_location (desktop_item), NULL, NULL);
 
-	error = NULL;
-
-	if (!g_spawn_sync (NULL, argv, NULL, G_SPAWN_SEARCH_PATH, NULL, NULL, &package_name, NULL,
-			&retval, &error))
-	{
-		g_warning ("error: [%s]\n", error->message);
-
-		return NULL;
-	}
-
-	g_free (argv[4]);
-	g_free (argv);
-
-	if (error)
-		g_error_free (error);
-
-	if (!retval)
-		return package_name;
-	else
-		return NULL;
 }
 
 gboolean
