--- libslab/application-tile.c	2006-10-03 13:38:50.000000000 -0600
+++ libslab/application-tile.c.new	2006-10-26 12:56:56.000000000 -0600
@@ -360,7 +360,7 @@
 	}
 
 /* make upgrade action */
-
+/*
 	if (verify_package_management_command (SLAB_UPGRADE_PACKAGE_KEY)) {
 		action = tile_action_new (TILE (this), upgrade_trigger, _("Upgrade"), 0);
 		actions [APPLICATION_TILE_ACTION_UPGRADE_PACKAGE] = action;
@@ -368,7 +368,6 @@
 		gtk_container_add (menu_ctnr, menu_item);
 	}
 
-/* make uninstall action */
 
 	if (verify_package_management_command (SLAB_UNINSTALL_PACKAGE_KEY)) {
 		action = tile_action_new (TILE (this), uninstall_trigger, _("Uninstall"), 0);
@@ -376,7 +375,7 @@
 		menu_item = GTK_WIDGET (tile_action_get_menu_item (action));
 		gtk_container_add (menu_ctnr, menu_item);
 	}
-
+*/
 	priv->gconf_client = gconf_client_get_default ();
 	priv->gconf_conn_id = gconf_client_notify_add (
 		priv->gconf_client, SLAB_USER_SPECIFIED_APPS_KEY,
