--- evolution-rss-0.1.2.orig/src/Makefile.am.orig	2009-06-12 00:51:40.000000000 +0200
+++ evolution-rss-0.1.2/src/Makefile.am	2009-06-12 00:52:02.000000000 +0200
@@ -91,26 +91,6 @@
 @INTLTOOL_SERVER_RULE@
 @INTLTOOL_SCHEMAS_RULE@
 
-install-data-local:
-	rm -f $(PLUGIN_INSTALL_DIR)/*gnome-cooly*
-	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-install-rule $(schema_DATA)
-
-uninstall-local:
-	GCONF_CONFIG_SOURCE=$(GCONF_SCHEMA_CONFIG_SOURCE) $(GCONFTOOL) --makefile-uninstall-rule $(schema_DATA)
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/remove_folder
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/pass_proxy
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/startup_check
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/rep_check
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/rep_check_timeout
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/auth_proxy
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/host_proxy
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/html_render
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/use_proxy
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/user_proxy
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/port_proxy
-	$(GCONFTOOL) -u /apps/evolution/evolution-rss/display_summary
-
-
 DISTCLEANFILES = $(schema_DATA)
 
 glade_DATA =                    \
