--- sickbeard/db.py.orig	2011-03-26 22:25:36.518731498 -0600
+++ sickbeard/db.py	2011-03-26 22:26:45.637867392 -0600
@@ -35,7 +35,7 @@
 
 		self.dbFileName = dbFileName
 
-		self.connection = sqlite3.connect(os.path.join(sickbeard.PROG_DIR, self.dbFileName), 20)
+		self.connection = sqlite3.connect(os.path.join(sickbeard.DATA_DIR, self.dbFileName), 20)
 		self.connection.row_factory = sqlite3.Row
 
  	def action(self, query, args=None):
