=== modified file 'firefox-libzg/extension/chrome.manifest'
--- firefox-libzg/extension/chrome.manifest	2010-05-11 15:51:49 +0000
+++ firefox-libzg/extension/chrome.manifest	2011-04-18 19:09:22 +0000
@@ -1,2 +1,4 @@
+binary-component	components/libzeitgeist_xpcom.so
+interfaces	components/zeitgeist.xpt
 content		zeitgeist				chrome/content/
 overlay		chrome://browser/content/browser.xul	chrome://zeitgeist/content/zeitgeist.xul

=== modified file 'firefox-libzg/extension/components/ZeitgeistComponent.cpp'
--- firefox-libzg/extension/components/ZeitgeistComponent.cpp	2010-05-12 06:44:01 +0000
+++ firefox-libzg/extension/components/ZeitgeistComponent.cpp	2011-04-18 18:34:35 +0000
@@ -20,20 +20,31 @@
  */
  
  
-#include "nsIGenericFactory.h"
+#include "mozilla/ModuleUtils.h"
 #include "zeitgeist.h"
 #include "zeitgeistextend.h"
 
 NS_GENERIC_FACTORY_CONSTRUCTOR(zeitgeistextend)
 
-static nsModuleComponentInfo components[] =
-{
-  {
-    ZEITGEIST_COMPONENT_CLASSNAME,
-    ZEITGEIST_COMPONENT_CID,
-    ZEITGEIST_COMPONENT_CONTRACTID,
-    zeitgeistextendConstructor,
-  }
-};
-
-NS_IMPL_NSGETMODULE("ZeitgeistModule", components)
+NS_DEFINE_NAMED_CID(ZEITGEIST_COMPONENT_CID);
+
+static const mozilla::Module::CIDEntry kZeitgeistCIDs[] = {
+  { &kZEITGEIST_COMPONENT_CID, false, NULL, zeitgeistextendConstructor },
+  { NULL }
+};
+
+static const mozilla::Module::ContractIDEntry kZeitgeistContracts[] = {
+  { ZEITGEIST_COMPONENT_CONTRACTID, &kZEITGEIST_COMPONENT_CID },
+  { NULL }
+};
+
+static const mozilla::Module kZeitgeistModule = {
+  mozilla::Module::kVersion,
+  kZeitgeistCIDs,
+  kZeitgeistContracts,
+  NULL
+};
+
+NSMODULE_DEFN(ZeitgeistModule) = &kZeitgeistModule;
+
+NS_IMPL_MOZILLA192_NSGETMODULE(&kZeitgeistModule)

