--- test/module.rules	2009-08-16 23:50:29.023024080 -0600
+++ test/module.rules.new	2009-08-17 00:00:38.614068434 -0600
@@ -25,7 +25,8 @@
 ifneq ($(BUILD.system),darwin)
 
 test.install: | $(dir $(TEST.install.exe))
-	$(CP.exe) $(TEST.exe) $(TEST.install.exe)
+	mkdir -p $(DESTDIR)/$(dir $(TEST.install.exe))
+	$(CP.exe) $(TEST.exe) $(DESTDIR)/$(TEST.install.exe)
 
 test.uninstall:
 	$(RM.exe) -f $(TEST.install.exe)
