--- gtk/src/Makefile.am	2009-08-16 23:00:11.572044284 -0600
+++ gtk/src/Makefile.am.new	2009-08-16 23:03:09.297014967 -0600
@@ -171,7 +171,7 @@
 		mkdir -p $(DESTDIR)/$(datadir)/icons/hicolor/$$SIZE/apps/; \
 		$(INSTALL_DATA) $(srcdir)/$$icon $(DESTDIR)/$(datadir)/icons/hicolor/$$SIZE/apps/$$FILE; \
 	done
-	@-if test -n "$(DESTDIR)/$(datadir)"; then \
+	@-if test -z "$(DESTDIR)"; then \
 		echo "Updating Gtk icon cache."; \
 		$(gtk_update_icon_cache); \
 	else \
