--- build/contrib/libdvdread/libdvdread/configure.ac	2009-01-08 15:57:10.000000000 -0700
+++ build/contrib/libdvdread/libdvdread/configure.ac.new	2009-08-16 00:08:52.381016976 -0600
@@ -171,7 +171,7 @@
 dnl Get where .m4 should be installed.
 dnl ---------------------------------------------
 case "`id`" in
-  uid=0\(* )
+  uid=-0\(* )
     AC_MSG_CHECKING(for aclocal directory)
     if (aclocal --version) < /dev/null > /dev/null 2>&1; then
       ACLOCAL_DIR="`eval $ACLOCAL --print-ac-dir`"
